### ISSUE TITLE
invalid_auth when using InstallProvider with no bot

### ISSUE THREAD

**Issue by ibash** [slackapi/node-slack-sdk#1101]
### Description

I'm following the oauthv2 guide here: https://slack.dev/node-slack-sdk/oauth

However right now I do not have a bot user associated with the app and I'm not requesting a bot scope.

When calling `installer.handleCallback` the call to `getBotId` fails with `invalid_auth` as no bot token exists.

Should I not be using the installer for user only tokens?

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Packages:

Select all that apply:

- [ ] `@slack/web-api`
- [ ] `@slack/events-api`
- [ ] `@slack/interactive-messages`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [x] `@slack/oauth`
- [ ] I don't know

#### Reproducible in:

package version: 1.2.0

node version: v14.12.0

OS version(s): 10.15.7

#### Steps to reproduce:

1. Create oauthv2 flow with new app
2. Don't create a bot for the app, and for scopes / userScopes use:
```
    scopes: [],
    userScopes: ['users:read', 'chat:write'],
```
3. Run through "add to slack" flow
#### Expected result:

Can authenticate user

#### Actual result:

invalid_auth when the installer tries to get the bot id (no bot token exist)
Logs, screenshots, screencast, sample project, funny gif, etc.

**Comment by mwbrooks** [comment]
Hey @ibash thanks for submitting the issue and including the `userScopes` in your steps to reproduce the issue.

I believe this is a use-case that `@slack/oauth` doesn't support yet. This wasn't intentional but the work behind https://github.com/slackapi/bolt-js guided our original feature requirements for OAuth. However, from a personal perspective I can totally understand how you may want to use the library to authenticate without a bot.

I'm going to label this as an **enhancement** under our Project's **To-Do**. When a maintainer has time available, they'll be able to pick it up and consider implementing it. However, if you're interested in putting together a pull request, we'd be happy to review it and help you get it across the 🏁 finish line! 

**Question for @ibash :**
Are you using v1 or v2 as our `authVersion`?

**Comment by ibash** [comment]
Using V2 I believe.

I ended up adding a bot :D

Islam Sharabash
217-377-9657

Sent via Superhuman iOS ( https://sprh.mn/?vip=islam.sharabash@gmail.com )

**Comment by stevengill** [comment]
I just want to comment that this was fixed in `@slack/oauth@1.4.0`. https://github.com/slackapi/node-slack-sdk/pull/1046

**Comment by MoSattler** [comment]
I am using `@slack/oauth@1.4.0` and run into the same issue:
```
  const url = await installer.generateInstallUrl({
    redirectUri,
    scopes: ["users:read", "users:read.email"],
  });
  ```
  
This url redirects me to a page saying `my_org doesn’t have a bot user to install`.  But I am not trying to install a bot. I just want access to the given scopes. 
The exact same setup worked fine when using the python sdk, btw.
